rosidl_generator_cpp/resource/msg__struct.hpp.em

```diff
@@ -149,31 +149,46 @@ def generate_zero_string(membset, fill_args):
 @[if not member_list]@
     (void)_init;
 @[end if]@
-@[for membset in member_list]@
-@[ if membset.members[0].default_value is not None]@
+@{
+members_if_all_defaults = [m for m in member_list if m.members[0].default_value]
+members_else_if_zero = [m for m in member_list if m.members[0].zero_value]
+members_if_all_zero = [
```

Comment:
I find the variable names pretty confusing. I would suggest to remove the `if` / `all` / `else` parts and only name the variable after what it contains and not how it is being used below.

Reply:
Yeah, that's a good point. Fixed.